### ISSUE TITLE
Need a hipster idea for Crystal website

### ISSUE THREAD

**Issue by jamesfuture** [crystal-lang/crystal-website#68]
My first impression on Crystal Lang website (https://crystal-lang.org) feels dry, boring and too many code examples in a page which can do more harm or scared away newbie or non-IT savvy when they visit the site for the first time, in contrast to RubyOnRails, Laravel, Vapor, etc.

If we want to grow community from all walk of life, we’ll need to revamp with friendly introductions and what made Crystal so special from other languages, this way the community will adopt without feeling difficulty or less motivated to pick up 2nd language after Ruby.

What do you guys think or some suggestions could be interesting.

**Comment by straight-shoota** [comment]
I actually like to have code examples, but the ones currently available could be more concise. Especially the one for C bindings. That's just *way* to much information for an introductory page.

**Comment by bowyan** [comment]
Maybe it might be possible to create a "Code Example" page. 

Left side navigation with: 

- Command line arguments
- Http Server
- ...

Center page with the examples and some basic text what the function does.

With categories and a mix of code examples.

And we simply point the people after 2 or 3 examples from the home page, to this sub page. I personally keep over 80 examples in my coding directory simply for quick code lookup for when i forget something.

The documentation and API for Crystal is a bit messy so i resort to this to keep track of the most common used code pieces.

**Comment by bcardiff** [comment]
I would like to see some small guides/snippet grouped by area of interest: Web Servers, Web Clients, CLI, Data Science, GUI, NLP, Gaming, etc. Although the could be linked from the website I think they would fit best in the "Overview and Examples" or a new section in https://github.com/crystal-lang/crystal-book right after installation.

**Comment by RX14** [comment]
The frontpage is exactly perfect as is. Please do not change it to remove code samples.

It has documentation links right at the top of the page, and having many code examples is good to evaluage the language.

The further we are from any "hipster ideas" the better. The current website even looks nice without JS, heck it even looks alright in `lynx`.

**Comment by RX14** [comment]
I agree that the C bindings example should be shortened, and the long-running issue about not having any tutorial-style introductory documentation stands, but the website currently does an excellent job at explaining how crystal is so special and I see no point in changing the website a lot.

**Comment by vladfaust** [comment]
Hipster ideas would lead to wider coverage and shareabiliity. JS is popular because its libs are hipsty and modern-designed (visually). Not everyone in the world uses text-based browsers and IRC, most select products by its visual representation and ease of use. Stop ludditting, Crystal is modern inside, so let it be outside.

**Comment by RX14** [comment]
@vladfaust the website is already modern in design, *and* accessible. This is not a dichotomy.

**Comment by vladfaust** [comment]
I'm just pretending to understand things 🤔

**Comment by bowyan** [comment]
Let me disagree with you RX14...

On the front page:

- The logo is taking up already half the page. That is content where you can show details about the language.

- The blog is hidden at the bottom of the page. A spot where nobody is going to look for it. News is supposed to be front and center, not hidden at the bottom.

- There is totally no need to show all the release notes for the old versions. People are interested in the latest release. Very few people are going to look up the past release note, and if they do, put them in a sub-page where you can download Crystal.

- A 48 Pixel "Latest release 0.26.1" is just bad design. And you do realize that focusing on **0.**xx that you are telling people. Do not use our language because its not API stable or even fully released.

- Press any menu item, the entire menu now slides from center to right side. Totally no integration between front and sub-pages.

- You do not need a 72px blog title, let alone a 465px "header". 

- The list goes on and on... I have not even started talking about the API and Docs layout what is just bad. Purple color, 50 page long API descriptions, ...

Yes, the website is not "bad" but lets be honest and not call it great. Modern does not mean trow everything in a 8 page lengths long front-page. The website is designed as a page filler because its lacking content.

Lets take one of your competitors:

https://dlang.org/

- Quick summery on the top what the language is about.
- Code example instantly viable without scrolling.
- News instantly visable
- Going to sub-pages keeps the menu structure intact
- Latest release and changelog is combined and 18px instead of 72px...

Yes, its not as "modern" but its more practical. And i have a **lot** of issues with the D site but its way more informative then Crystal.

Call it my personal opinion but a lot of programmers are square head people who do not realize how perception shapes how people thing about a product. And yes, Crystal is a product. 

Take a look at the Inko frontpage:

https://inko-lang.org/

Simple, gets the point across. It can use a news/blog section but its just a new project. People congratulate the developer for having a clear and interesting website, that puts the code / language first. Hell, it even tweaked my interest just from the website and the instant code example.

Somebody on reddit mentioned that Crystal doc sucks... and he pointed out how most people look up information. Namely via Google: https://www.google.com/search?q=crystal+language+date

I see version 0.20 API, then 0.18 API and nothing interesting after that... And yes, its 50 pages on my 1080p screen for the date API. That is just crazy.

Crystal has a lot of issues on its website, doc, API and trying to deny it RX14 is only going to cost Crystal people, not gain them. I can nitpick about these issues but if core developers deny the issues, i fear we have nothing to talk about.

**Comment by proyb6** [comment]
@bowyan We could certainly agree not many would spent time reading home page, it could be divide to different categories that’s friendly for mobile.

I must say Scala is well designed
https://www.scala-lang.org/

**Comment by RX14** [comment]
@bowyan thanks for the specific feedback! In fact, it seems that you want to take the page in a more information-dense "less hipstery" way, and I agree with a lot of your points. My apologies for calling it "really good", it's probably closer to "pretty good".

- I actually like the header, and it's styling. If anything I'd move the "latest release" part to push up the examples.
- I'd rather show examples before the blog, after all the blog link is in the header nav.
- I think showing a release history with dates is a good idea to show that the project is active.
- I disagree that we should hide that crystal is 0.x, but I don't think it's the greatest use of above-the-fold pixels either.
- Agreed.
- I think the layout of the blog pages is fine, and it keeps the first blog post/paragraph above the fold.


The API and docs are indeed definitely lacking, although that's a different issue for a different repo. I think they need *far* more improvement than the frontpage.

**Comment by proyb6** [comment]
Let also build a better ecosystem with Crystal!

**Comment by straight-shoota** [comment]
@bowyan I'd suggest to focus on the front page here to not get lost in too many details. That's the most prominent page. API docs and language reference are entirely different issues and sub pages can also be addressed later.

But I agree on you criticism of the front page:

1. Header is too large. The top of the page should provide more information.
2. Articles should be more prominent.
3. Release notes history is unnecessary.
4. Latest release is to strong and should be coupled with the install button.
5. Needs a proper introduction/description of the language.
6. Improved and simplified code examples.

A quick draft for an improved header: https://codepen.io/straight-shoota/pen/MqGezx

**Comment by RX14** [comment]
Making the actual `.header-section` smaller looks ugly, and so does splitting it into two columns. I'm skeptical that a better design can be found there without redesigning the whole page, which I feel is a waste of effort.

**Comment by straight-shoota** [comment]
@RX14 I think the two columns in the header actually look quite neat.

**Comment by straight-shoota** [comment]
We have just launched a major overhaul of the website (#416), incorporating some ideas from this thread.
